src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -955,8 +971,8 @@ public String toString() {
       // Unit tests don't want the extra data.  (Maybe we should fix the goldens?)
       String clazzName = this.getClass().getSimpleName();
       String text = getCanonicalText();
-      ss += ":" + (null == text ? "<no text>" : text);
-      ss += ":" + (null == clazzName ? "<anonymous>" : clazzName);
+      ss += ":" + text;
```

Comment:
Why?  Can text and clazzName no longer be null?

Reply:
I haven't found any override of getClass, or getSimpleName, and inherited implementation never returns null. getCanonicalText marked as `@NotNull`.
I have no idea why is it should return nulls.

Reply:
I missed this the first time:  Besides dropping the null test,  this also drops the class name from the output.  Why do we want to do that?  If we really want to drop it, then we should remove the clazzname variable too.  But I would rather keep it, personally.

Reply:
It turns out that clazzname won't come back null -- it will come back empty in the case of anonymous.  We still need the line, but a different test.

Reply:
https://github.com/HaxeFoundation/intellij-haxe/issues/732